mono/mini/interp/interp.c

```diff
@@ -4545,7 +4545,19 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 			}
 
 			if (frame->imethod->method->wrapper_type != MONO_WRAPPER_RUNTIME_INVOKE) {
-				MonoException *abort_exc = mono_thread_get_undeniable_exception ();
+				stackval tmp_sp;
+
+				child_frame.parent = frame;
+				/*
+				 * We need for mono_thread_get_undeniable_exception to be able to unwind
+				 * to check the abort threshold. For this to work we use child_frame as a
+				 * dummy frame that is stored in the lmf and serves as the transition frame
+				 */
```

Comment:
good call on pushing a LMF here, but why do we need a dummy frame?  shouldn't be the current interp frame enough?

Reply:
The frame that is pushed with LMF is unwinded by EH as a special cased transition frame (FRAME_TYPE_INTERP_TO_MANAGED). The current frame is just a normal managed frame and should be a FRAME_TYPE_INTERP. Added the dummy frame to keep this logic.

Alternatively we could teach EH to unwind directly to an INTERP frame

Reply:
maybe I'm missing something, but it sounds like we are missing this kind of dummy frame in a couple of other places too.